File: daily/739/solution.js
// 题目来源：https://leetcode-cn.com/problems/daily-temperatures/

var dailyTemperatures = function(T) {
    var n = T.length;

    var ans = new Array(n).fill(0);
    var stack = [];
    var top = -1;
    for(var i = 0; i < n; i++) {
        var t = T[i];
        while(stack.length > 0 && T[stack[top]] < t) {
            ans[stack[top]] = i - stack[top];
            stack.pop();
            top--;
        }
        stack.push(i);
        top++;
    }

    return ans;
};

File: daily/739/solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        int n = T.size();
        vector<int> ans(n);
        stack<int> s;
        for(int i = 0; i < n; ++i){
            while(!s.empty() && T[i] > T[s.top()]){
                int preIndex = s.top();
                ans[preIndex] = i - preIndex;
                s.pop();
            }
            s.push(i);
        }
        return ans;
    }
};